src/platform_impl/apple/appkit/view.rs

```diff
@@ -639,19 +639,32 @@ declare_class!(
         }
 
         #[method(mouseEntered:)]
-        fn mouse_entered(&self, _event: &NSEvent) {
+        fn mouse_entered(&self, event: &NSEvent) {
             trace_scope!("mouseEntered:");
-            self.queue_event(WindowEvent::CursorEntered {
+
+            let view_point = self.mouse_view_point(event);
+
+            if !self.is_in_window(view_point) {
+                return
+            }
+
```

Comment:
Nit: Redundant check (if the mouse just entered the view, then it cannot be outside of it).

```suggestion
```